lib/improver/tests/nowcasting/optical_flow/test_AdvectField.py

```diff
@@ -253,6 +256,35 @@ def test_basic(self):
         result = self.plugin.process(self.cube, self.timestep)
         self.assertIsInstance(result, iris.cube.Cube)
 
+    def test_metadata(self):
+        """Test plugin returns a cube with the desired metadata."""
+        metadata_dict = {"attributes": {
+                            "mosg__grid_version": "1.0.0",
+                            "mosg__model_configuration": "nc_det",
+                            "source": "Met Office Nowcast",
+                            "institution": "Met Office",
+                            "title": "Nowcast on UK 2 km Standard Grid"}}
+        plugin = AdvectField(self.vel_x, self.vel_y,
+                             metadata_dict=metadata_dict)
+        result = plugin.process(self.cube, self.timestep)
+        result.attributes.pop("history")
+        self.assertEqual(result.attributes, metadata_dict["attributes"])
+
+    def test_check_source_metadata(self):
```

Comment:
I don't know if you need a test for the case where you don't have the "institution" attribute on your cube as well.

Reply:
Added a test.